sphinxdocs/private/sphinx_server.py

```diff
@@ -28,5 +31,19 @@ def __init__(self, *args, **kwargs):
     return 0
 
 
+@contextlib.contextmanager
+def _start_server(handler, ip, start_port):
+    for port in range(start_port, start_port + 10):
```

Comment:
nit: an alternative would be to open a socket on a random port and then close it and then use the previously allocated port. It's a little racy, but could work most of the time. That said, this is probably better.